Apps/Gateway/src/ClientApp/src/stores/user.ts

```diff
@@ -0,0 +1,468 @@
+        user.value.hasEmail = !!userProfile?.email;
+        user.value.verifiedEmail = userProfile.isEmailVerified;
+        user.value.hasSMS = !!userProfile?.smsNumber;
+        user.value.verifiedSMS = userProfile.isSMSNumberVerified;
+        user.value.hasTourUpdated = userProfile.hasTourUpdated ?? false;
+
+        logger.verbose(`User: ${JSON.stringify(user.value)}`);
+
+        setLoadedStatus(patient.value.hdid !== undefined);
+    }
+
+    function setSmsResendDateTime(date: DateWrapper) {
+        smsResendDateTime.value = date;
+        setLoadedStatus();
+    }
+
+    function setUserError(errorMessage: string) {
+        error.value = true;
+        statusMessage.value = errorMessage;
+        status.value = LoadStatus.ERROR;
+    }
+
+    function setLoadedStatus(isCompletelyLoaded: boolean = true) {
+        error.value = false;
+        statusMessage.value = "Success";
+        status.value = isCompletelyLoaded
+            ? LoadStatus.LOADED
+            : LoadStatus.PARTIALLY_LOADED;
+    }
+
+    function bindUserPreference(userPreference: UserPreference) {
+        logger.debug(
+            `setUserPreference: preference.name: ${JSON.stringify(
+                userPreference.preference
+            )}, preference.value: ${JSON.stringify(userPreference.value)}`
+        );
+        user.value.preferences = Object.assign({}, user.value.preferences, {
+            [userPreference.preference]: userPreference,
+        });
+        setLoadedStatus();
+    }
+
+    function setPatient(incomingPatient: Patient) {
+        logger.debug(`setPatient: ${JSON.stringify(patient)}`);
+        patient.value = incomingPatient;
+        setLoadedStatus(user.value.hdid !== undefined);
+    }
+
+    function handleError(
+        resultError: ResultError,
+        errorType: ErrorType,
+        errorSource: ErrorSourceType
+    ) {
+        logger.error(`Error: ${JSON.stringify(resultError)}`);
+        setUserError(resultError.resultMessage);
+
+        if (resultError.statusCode === 429) {
+            const errorKey = "page";
+            if (errorType === ErrorType.Retrieve) {
+                errorStore.setTooManyRequestsWarning(errorKey);
+            } else {
+                errorStore.setTooManyRequestsError(errorKey);
+            }
+        } else {
+            errorStore.addError(errorType, errorSource, resultError.traceId);
+        }
+    }
+
+    function createProfile(request: CreateUserRequest): Promise<void> {
+        return new Promise((resolve, reject) =>
+            userProfileService
+                .createProfile(request)
+                .then((userProfile) => {
+                    logger.verbose(
+                        `User Profile: ${JSON.stringify(userProfile)}`
+                    );
+                    setProfileUserData(userProfile);
+                    resolve();
+                })
+                .catch((resultError: ResultError) => {
+                    setUserError(resultError.resultMessage);
```

Comment:
```suggestion
                    handleError(
                        resultError,
                        ErrorType.Create,
                        ErrorSourceType.Profile
                    );
```

This used to dispatch the handleError action, not the userError mutation.